crates/typos-dict/tests/verify.rs

```diff
@@ -74,14 +69,16 @@ fn generate<W: std::io::Write>(file: &mut W, dict: &[u8]) {
             }
         })
         .map(|(typo, corrections)| {
-            let mut new_corrections = BTreeSet::new();
-            for correction in corrections {
-                let correction = word_variants
-                    .get(correction.as_str())
-                    .and_then(|words| find_best_match(&typo, correction.as_str(), words))
-                    .unwrap_or(&correction);
-                new_corrections.insert(correction.to_owned());
-            }
+            let new_corrections: BTreeSet<_> = corrections
+                .into_iter()
+                .map(|correction| {
+                    word_variants
+                        .get(&*correction)
+                        .and_then(|variants| find_best_match(&typo, correction.as_str(), variants))
+                        .map(|s| s.to_owned())
+                        .unwrap_or(correction)
+                })
+                .collect();
```

Comment:
imo this is not a definitive enough of an improvement to make

Relying on combinators too much can make the code harder to read, debug, etc.

Reply:
It's a preliminary refactor for ce3da1f38cc0a068ae433701067d0af320c9f347 of #765, which changes the BTreeSet to a Vec ... with the combinator approach duplicates can then simply be removed by adding a `.unique()` from Itertools, which I think is a bit more elegant than coding the hashset explicitly.

Reply:
I'll also add overuse of combinators make refactors harder.

Relying further on a bad change doesn't make it better.

(that use of `unique` is also incorrect, its in the wrong place.  Switching to an `IndexSet` is a more trivial change and would be correct)

Reply:
Fair enough ... I've now also dropped this change.

I was hesitant to add a new dependency on indexmap but I guess we already indirectly depend on it via the human-panic dependency of typos-cli.

Reply:
We also directly depend on `toml` which depends on `toml_edit` which gives us `indexmap`